src/lib/utils/meta_tables/meta_system_utilization_table.cpp

```diff
@@ -0,0 +1,270 @@
+}
+
+/*
+ * Returns the time in ns that ALL processes have spent on the CPU
+ * since an arbitrary point in the past.
+*/
+uint64_t MetaSystemUtilizationTable::_get_system_cpu_time() {
+#ifdef __linux__
+  std::ifstream stat_file;
+  stat_file.open("/proc/stat", std::ifstream::in);
+  DebugAssert(stat_file.is_open(), "Failed to open /proc/stat");
+
+  std::string cpu_line;
+  std::getline(stat_file, cpu_line);
+  stat_file.close();
+
+  const auto cpu_ticks = _parse_value_string(cpu_line);
+
+  const auto user_ticks = cpu_ticks.at(0);
+  const auto user_nice_ticks = cpu_ticks.at(1);
+  const auto kernel_ticks = cpu_ticks.at(2);
+
+  const auto active_ticks = user_ticks + user_nice_ticks + kernel_ticks;
+  size_t cpu_count;
+  if (numa_available() != -1) {
+    cpu_count = numa_num_task_cpus();
+  } else {
+    cpu_count = _get_cpu_count();
+  }
+
+  // The amount of time in /proc/stat is measured in units of clock ticks.
+  // sysconf(_SC_CLK_TCK) can be used to convert it to ns.
+  const auto active_ns = (active_ticks * std::nano::den) / (sysconf(_SC_CLK_TCK) * cpu_count);
+
+  return active_ns;
+#endif
+
+#ifdef __APPLE__
+  host_cpu_load_info_data_t cpu_info;
+  mach_msg_type_number_t count = HOST_CPU_LOAD_INFO_COUNT;
+  [[maybe_unused]] const auto ret =
+      host_statistics(mach_host_self(), HOST_CPU_LOAD_INFO, reinterpret_cast<host_info_t>(&cpu_info), &count);
+  DebugAssert(ret == KERN_SUCCESS, "Failed to get host_statistics");
+
+  const auto active_ticks =
+      cpu_info.cpu_ticks[CPU_STATE_SYSTEM] + cpu_info.cpu_ticks[CPU_STATE_USER] + cpu_info.cpu_ticks[CPU_STATE_NICE];
+
+  // The amount of time from HOST_CPU_LOAD_INFO is measured in units of clock ticks.
+  // sysconf(_SC_CLK_TCK) can be used to convert it to ns.
+  const auto active_ns = active_ticks * std::nano::den / (sysconf(_SC_CLK_TCK) * _get_cpu_count());
+
+  return active_ns;
+#endif
+
+  Fail("Method not implemented for this platform");
+}
+
+/*
+ * Returns the time in ns that THIS process has spent on the CPU
+ * since an arbitrary point in the past.
+*/
+uint64_t MetaSystemUtilizationTable::_get_process_cpu_time() {
+  // CLOCK_PROCESS_CPUTIME_ID:
+  // Per-process CPU-time clock (measures CPU time consumed by all threads in the process).
+#ifdef __linux__
+  struct timespec time_spec {};
+  [[maybe_unused]] const auto ret = clock_gettime(CLOCK_PROCESS_CPUTIME_ID, &time_spec);
+  DebugAssert(ret == 0, "Failed in clock_gettime");
+
+  size_t cpu_count;
+  if (numa_available() != -1) {
+    cpu_count = numa_num_task_cpus();
+  } else {
+    cpu_count = _get_cpu_count();
+  }
+
+  const auto active_ns = (time_spec.tv_sec * std::nano::den + time_spec.tv_nsec) / cpu_count;
+
+  return active_ns;
+#endif
+
+#ifdef __APPLE__
+  const auto active_ns = clock_gettime_nsec_np(CLOCK_PROCESS_CPUTIME_ID);
+  DebugAssert(active_ns != 0, "Failed in clock_gettime_nsec_np");
+
+  return active_ns / _get_cpu_count();
+#endif
+
+  Fail("Method not implemented for this platform");
+}

```

Comment:
Depending on the outcome of the previous discussion, this might be more accurate:

```
struct task_basic_info {
        integer_t       suspend_count;  /* suspend count for task */
        vm_size_t       virtual_size;   /* number of virtual pages */
        vm_size_t       resident_size;  /* number of resident pages */
        time_value_t    user_time;      /* total user run time for
                                           terminated threads */
        time_value_t    system_time;    /* total system run time for
                                           terminated threads */
	policy_t	policy;		/* default policy for new threads */
};
```

Reply:
Guess this is resolved then.